src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -209,10 +210,22 @@
                             (flatten-row xs path))))))]
     (into {} (flatten-row row [field-name]))))
 
+(defn- type-finder
```

Comment:
I think this function name is a little confusing. If I see `(type-finder x)` it's not obvious at all what it does... I'd almost expect `type-finder` to return a type-finding function or something. I'm struggling to find a good name to accurately describe what it does... maybe `type-by-parsing-datetime-strings` or something like that. IDK.

Another option is giving it a name describing what it's for. Maybe `type-for-json-xform` or whatever makes more sense